pkg/monitor/clustermonitor.go

```diff
@@ -229,13 +234,29 @@ func (m *Monitor) deleteCluster(managedCluster *clusterv1.ManagedCluster) {
 
 // GetLocalCluster Get Local cluster ID
 func (m *Monitor) GetLocalCluster() string {
-	lock.RLock()
-	defer lock.RUnlock()
+	var clusterVersionGvr = schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: "clusterversions",
+	}
 	glog.V(2).Info("Get Local Cluster ID.")
-	for _, cluster := range m.ManagedClusterInfo {
-		if "local-cluster" == cluster.Namespace {
-			return cluster.ClusterID
-		}
+	dynamicClient := config.GetDynamicClient()
+	versionObj, err := dynamicClient.Resource(clusterVersionGvr).Get(context.TODO(), "version", metav1.GetOptions{})
+	if err != nil {
+		glog.V(2).Infof("Failed to get clusterversions : %v", err)
+		return "-1"
+	}
+	clusterID, _, err := unstructured.NestedString(versionObj.Object, "spec", "clusterID")
+	if err != nil {
+		glog.V(2).Infof("Failed to get OCP clusterID from version: %v", err)
+		return "-1"
 	}
-	return "-1"
+	lock.Lock()
+	m.ManagedClusterInfo = append(m.ManagedClusterInfo, types.ManagedClusterInfo{
```

Comment:
We only want insights for clusters that are Openshift & version 4.X. Using this resource we can assume it is an Openshift cluster, but we will need to check the version is correct.

Reply:
Looks like we could use `status.desired.version`

Reply:
status.desired.version - not sure if thats the current version or the one RH. wants you to be in .

Reply:
`ACM is not support on hub on openshift 3.x`. Dont need to check version if we dont support less than 4.X. resolving

Reply:
Good point. Do we check this for the managed clusters?

Reply:
Yep we check the version for all managed clusters